pandas/core/computation/expr.py

```diff
@@ -39,8 +36,69 @@
 
 import pandas.io.formats.printing as printing
 
+# A token value Python's tokenizer probably will never use.
+_BACKTICK_QUOTED_STRING = 100
+
+Tok = Tuple[int, str]
+
+
+def _clean_special_characters_column_name(name: str) -> str:
+    """Check if name contains any special characters, if it contains any
+    special characters the special characters will be replaced by an special
+    string and an underscore suffix is added."""
+    if not isinstance(name, str):
+        return name
+
+    if name.isidentifier() and not iskeyword(name):
+        return name
+
+    # Create a list with the special characters and their replacement.
+    # So far we only replace single character operators.
+    special_characters_replacements = {
+        " ": "_",
+        **{
+            char: "_" + token.tok_name[tokval] + "_"
+            for char, tokval in tokenize.EXACT_TOKEN_TYPES.items()
+            if len(char) == 1
+        },
+    }
+
+    name = "".join(special_characters_replacements.get(char, char) for char in name)
+    name = "BACKTICK_QUOTED_STRING_" + name
+
+    # By my knowlegde the current conversion cannot fail, but who knows.
```

Comment:
Hmm do we have a test case that hits this?

Reply:
Well, I just cannot imagine something that could go wrong. We have unit tests that test things what came to mind, but they work. I cannot come up with a test case that tests if it would fail, because than I probably made a workaround for this. This exception is more there for the case that someone in the future by accident finds such a case, and then he/she gets a readable exception telling that raising an issue can help us improve this feature.

Reply:
I guess if we can't think of a way to actually hit the code branch we should just remove; speculating that this could get hit doesn't help too much

Reply:
I see. 

However, I found some way this can be hit. 

You can for example use emoticons as column names. There is now an test case that tests this to raise an exception. 

I don't think I will come up with something that can actually work with emoticons as column names. So, let's rest that for now. 

(I might be able come up with something, but let's just say that we just shouldn't go into such craziness.)